Tests/DatadogTests/Datadog/Core/Upload/DataUploaderTests.swift

```diff
@@ -46,6 +46,15 @@ class DataUploadURLProviderTests: XCTestCase {
         dateProvider.advance(bySeconds: 9.999)
         XCTAssertEqual(urlProvider.url, URL(string: "https://api.example.com/v1/endpoint/abc?ddsource=ios&batch_time=1576404009999"))
     }
+
+    func testIllegalCharactersInQueryItemsDontCrash() throws {
```

Comment:
Whitespaces are not "illegal" and the expected behaviour is not to "don't crash" (we could say so for all tests). To be more precise:
```suggestion
    func testItEscapesWhitespacesInQueryItems() throws {
```